src/api/app.js

```diff
@@ -69,10 +69,7 @@ module.exports.authorise = (appToken, permissions, options) => {
 module.exports.connectAuthorised = (appToken, authUri) => {
   return getObj(appToken)
     .then((app) => app.auth.loginFromURI(authUri))
-    .then((connectedApp) => {
-      freeObj(appToken);
-      return genHandle(connectedApp);
-    });
+    .then((connectedApp) => appToken);
```

Comment:
Thanks a lot @gnunicorn .
That other function you referenced @shankar2105 , we could eventually also expose it as a separate function so the client app can connect using the URI and the appInfo without needing to call `initialise` and `connectAuthorised` but just call this function.
So I think we should be good to apply this fix now.